src/foam/nanos/approval/ApprovalDAO.java

```diff
@@ -0,0 +1,108 @@
+package foam.nanos.approval;
+
+import foam.core.FObject;
+import foam.core.X;
+import foam.dao.DAO;
+import foam.dao.ProxyDAO;
+import foam.mlang.sink.Sum;
+import foam.nanos.auth.User;
+import foam.nanos.ruler.Operations;
+import foam.nanos.approval.RoleApprovalRequest;
+import foam.nanos.approval.ApprovalStatus;
+import foam.nanos.approval.ApprovalRequest;
+
+import static foam.mlang.MLang.*;
+
+/**
+ * When a single request is approved, checks all other approved requests and calculates points.
+ * If points are sufficient ( >= requiredPoints ), removes all unused requests from approvalRequestDAO and
+ * Re-submits the object to its dao (daoKey), where a rule should be defined to determine further actions.
+ */
+public class ApprovalDAO
```

Comment:
Why is the DAO based and Rule based?

Reply:
This is from current approval system before Liquid, we used it so as to not cause much conflicts when building the RoleApprovalSystem for Liquid. However I do now think this can also be migrated to a rule since this code is pretty old.

Reply:
ok, in future PR.